core/trino-main/src/main/java/io/trino/cost/FilterProjectAggregationStatsRule.java

```diff
@@ -0,0 +1,107 @@
+import io.trino.sql.planner.plan.PlanNode;
+import io.trino.sql.planner.plan.ProjectNode;
+
+import java.util.Optional;
+
+import static com.google.common.collect.MoreCollectors.onlyElement;
+import static io.trino.SystemSessionProperties.isUnestimatedFilterCostApproximationEnabled;
+import static io.trino.cost.FilterStatsCalculator.UNKNOWN_FILTER_COEFFICIENT;
+import static io.trino.sql.planner.plan.Patterns.filter;
+import static java.util.Objects.requireNonNull;
+
+/**
+ * AggregationStatsRule does not have sufficient information to provide accurate estimates on aggregated symbols.
+ * This rule finds unestimated filters on top of aggregations whose output row count was estimated and returns
+ * estimated row count of (0.9 * aggregation output row count) for the filter.
+ */
+public class FilterProjectAggregationStatsRule
+        extends SimpleStatsRule<FilterNode>
+{
+    private static final Pattern<FilterNode> PATTERN = filter();
+
+    private final FilterStatsCalculator filterStatsCalculator;
+
+    public FilterProjectAggregationStatsRule(StatsNormalizer normalizer, FilterStatsCalculator filterStatsCalculator)
+    {
+        super(normalizer);
+        this.filterStatsCalculator = requireNonNull(filterStatsCalculator, "filterStatsCalculator cannot be null");
+    }
+
+    @Override
+    public Pattern<FilterNode> getPattern()
+    {
+        return PATTERN;
+    }
+
+    @Override
+    protected Optional<PlanNodeStatsEstimate> doCalculate(FilterNode node, StatsProvider sourceStats, Lookup lookup, Session session, TypeProvider types)
+    {
+        if (!isUnestimatedFilterCostApproximationEnabled(session)) {
+            return Optional.empty();
+        }
+        PlanNode nodeSource = resolveGroup(lookup, node.getSource());
+        AggregationNode aggregationNode;
+        if (nodeSource instanceof ProjectNode) {
+            ProjectNode projectNode = (ProjectNode) nodeSource;
+            if (!projectNode.isIdentity()) {
+                return Optional.empty();
+            }
+            PlanNode projectNodeSource = resolveGroup(lookup, projectNode.getSource());
+            if (!(projectNodeSource instanceof AggregationNode)) {
+                return Optional.empty();
+            }
+            aggregationNode = (AggregationNode) projectNodeSource;
+        }
+        else if (nodeSource instanceof AggregationNode) {
+            aggregationNode = (AggregationNode) nodeSource;
+        }
+        else {
+            return Optional.empty();
+        }
+
+        return calculate(node, aggregationNode, sourceStats, session, types);
+    }
+
+    private Optional<PlanNodeStatsEstimate> calculate(FilterNode filterNode, AggregationNode aggregationNode, StatsProvider statsProvider, Session session, TypeProvider types)
+    {
+        PlanNodeStatsEstimate filteredStats = filterStatsCalculator.filterStats(statsProvider.getStats(filterNode.getSource()), filterNode.getPredicate(), session, types);
```

Comment:
nit: we assume that PPD pushdown will move filters that are not aggregation related (or non-deterministc). Add a comment